### DISCUSSION TITLE
OPA in Kubernetes to block create load balancer except with specific annotation in .rego

### DISCUSSION THREAD

**Discussion by cmwatts1974** [open-policy-agent/community#153] [OPA and Rego]
How do I write my .rego file to deny creation of loadbalancer except with specific annotation in .rego file...see below what I have tried

package kubernetes.admission

import data.kubernetes.namespaces

import input.request.object.metadata.annotations as annotations

deny[msg] {
    input.request.kind.kind = "Service"
    input.request.operation = "CREATE"
    input.request.object.spec.type = "LoadBalancer"
    missing_required_annotations[msg]
}
missing_required_annotations[msg] {
    not annotations["service.beta.kubernetes.io/aws-load-balancer-scheme = "internal"] = internal
}

**Comment by anderseknert** [comment]
Answered [here](https://github.com/open-policy-agent/community/discussions/154).
